govcd/vdc.go

```diff
@@ -837,3 +837,35 @@ func (vdc *Vdc) buildNsxvNetworkServiceEndpointURL(optionalSuffix string) (strin
 
 	return hostname, nil
 }
+
+// Finds VM template using catalog name, vApp template name, VN name in template. Returns types.QueryResultVMRecordType
+func (vdc *Vdc) QueryVappVmTemplate(catalogName, vappTemplateName, vmNameInTemplate string) (*types.QueryResultVMRecordType, error) {
+
+	typeVm := "vm"
+	if vdc.client.IsSysAdmin {
+		typeVm = "adminVM"
+	}
+
+	results, err := vdc.QueryWithNotEncodedParams(nil, map[string]string{"type": typeVm,
+		"filter": "catalogName==" + url.QueryEscape(catalogName) + ";containerName==" + url.QueryEscape(vappTemplateName) + ";name==" + url.QueryEscape(vmNameInTemplate) +
+			";isVAppTemplate==true;status!=FAILED_CREATION;status!=UNKNOWN;status!=UNRECOGNIZED;status!=UNRESOLVED&links=true;",
```

Comment:
Is there a chance that two VMs have the same name but different status? If not, should we check the status after the query with a chance for a better error handling?

Reply:
We quering VM templates, not VMs.

Reply:
The query type is "vm", hence we are querying VMs. 
My question is whether it would be cleaner to check the status after the query rather than having this complex condition in the query itself.

Reply:
I use same approach as UI doing, so assume it's safe.

Reply:
In addition - is this filter as UI does it? Just being cautious about various states vCD has

Reply:
Yes, it is reversed engineered from UI.